echo.js

```diff
@@ -1,7 +1,46 @@
-function echo(){
-  // ...
+// // PART 1:
+// // echo();
+// // should print nothing
+
+function echo() {
+    console.log();
 }
 
 echo();
+
+//------------------------
+//PART 2:
+//echo('bla');
+// should print
+//
+// 'bla'
+
+function echo(x) {
+    console.log(x);
+}
+
+echo('bla');
+
+// //ALTERNATE SOLUTION:
+
+function echo() {
+    console.log(arguments[0]);
+}
+
 echo('bla');
-echo('foo', 'bar', 'baz');
+
+//------------------------
+//PART 3:
+//echo('foo', 'bar', 'baz');
+// should print
+//
+// 'foo'
+// 'bar'
+// 'baz'
+
+function echo() {
+    for (var x=0; x <= arguments.length - 1; x++)
+    console.log(arguments[x]);
+}
```

Comment:
True! Thank you!

On Thu, Oct 29, 2015 at 9:26 AM, Aidan Feldman <notifications@github.com>
wrote:

> In echo.js
> <https://github.com/advanced-js/echo/pull/50#discussion_r43383469>:
>
> >  echo('bla');
> > -echo('foo', 'bar', 'baz');
> > +
> > +//------------------------
> > +//PART 3:
> > +//echo('foo', 'bar', 'baz');
> > +// should print
> > +//
> > +// 'foo'
> > +// 'bar'
> > +// 'baz'
> > +
> > +function echo() {
> > +    for (var x=0; x <= arguments.length - 1; x++)
> > +    console.log(arguments[x]);
> > +}
>
> Good news: this version would work for all of them!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/advanced-js/echo/pull/50/files#r43383469>.
>